MoreLinq/MoveRange.cs

```diff
@@ -0,0 +1,85 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Take a range of the source sequence and moves it to another position. 
+        /// </summary>
+        /// <typeparam name="T">Type of the source sequence</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="oldIndex">The zero-based and original index where the range to be moved begins.</param>
+        /// <param name="count">The count of items to move.</param>
+        /// <param name="newIndex">The index where the specified range will be placed.</param>
+        /// <returns>The sequence with the specified range moved to the new position.</returns>
+        /// <remarks>
+        /// This operator uses deferred execution and streams its results.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// var result = Enumerable.Range(0, 6).MoveRange(3, 2, 0);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>{ 3, 4, 0, 1, 2, 5 }</c>.
+        /// </example>
+
+        public static IEnumerable<T> MoveRange<T>(this IEnumerable<T> source, int oldIndex, int count, int newIndex)
```

Comment:
This is about moving, `fromIndex` and `toIndex` go better than `oldIndex` and `newIndex`.